src/AjaxDynamicDataTable.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import DynamicDataTable from "./DynamicDataTable";
+import {AxiosInstance as axios} from "axios";
+
+class AjaxDynamicDataTable extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            rows: [],
+            currentPage: 1,
+            totalPages: 1,
+            orderByField: null,
+            orderByDirection: null,
+        };
+    }
+
+    componentDidMount() {
+        this.loadPage(1);
+    }
+
+    render() {
+
+        const {rows, currentPage, totalPages, orderByField, orderByDirection} = this.state;
+
+        return (
+            <DynamicDataTable
+                rows={rows}
+                currentPage={currentPage}
+                totalPages={totalPages}
+                orderByField={orderByField}
+                orderByDirection={orderByDirection}
+                changePage={this.changePage.bind(this)}
```

Comment:
Unfortunately calling `.bind` in the render function still causes the issue as it creates a new function each time it is called causing the component to re-render unnecessarily. You need to refactor out the bind methods to the constructor.

Reply:
Moved binding to constructor.